open-sphere-plugins/ogc-server/src/main/java/io/opensphere/server/control/AbstractServerSourceController.java

```diff
@@ -34,6 +43,14 @@
     /** The Change support. */
     private final WeakChangeSupport<ConfigChangeListener> myChangeSupport;
 
+    /** Listener for changes to the proxy settings. */
+    private final NetworkConfigurationManager.NetworkConfigurationChangeListener myNetworkConfigChangeListener =
+            this::reloadBadSources;
+
+    /** The Map of sources to reload listeners. */
+    private final Map<IDataSource, EventListener<AbstractSingleStateEvent>> myReloadListenerMap =
+            new HashMap<IDataSource, EventListener<AbstractSingleStateEvent>>();
```

Comment:
Use a synchronized map.

Reply:
Done. Also synchronized on the map where multiple operations were done at once - let me know if I still have synchronization mistakes I missed.